File: src/aoc2021/Day1/DepthIncreaseFinder.java
package aoc2021.Day1;

import aoc2020.utils.FileUtils;
import aoc2021.utils.NumberUtils;

import java.util.List;

public class DepthIncreaseFinder {

    private static String inputPath1 = "C:\\MyDrive\\Study\\My_Github_Repo\\Advent-of-Code\\src\\aoc2021\\Day1\\input1";
    private static String inputPath2 = "C:\\MyDrive\\Study\\My_Github_Repo\\Advent-of-Code\\src\\aoc2021\\Day1\\input2";

    public static void main(String[] args) {
        List<String> input1 = FileUtils.readInputFile(inputPath1);
        System.out.println(getIncreasedWindowDepthCount(input1));

        List<String> input2 = FileUtils.readInputFile(inputPath2);
        System.out.println(getIncreasedDepthCount(input2));
        System.out.println(getIncreasedWindowDepthCount(input2));
    }

    public static int getIncreasedDepthCount(List<String> input) {
        List<Integer> depths = NumberUtils.convertToInt(input);
        int result = 0;

        for (int i = 1; i < depths.size(); i++) {
            if (depths.get(i) > depths.get(i - 1)) {
                result++;
            }
        }
        return result;
    }

    public static int getIncreasedWindowDepthCount(List<String> input) {
        List<Integer> depths = NumberUtils.convertToInt(input);
        int result = 0;
        int prev = -1;
        int currentSum;
        for (int i = 0; i < depths.size() - 3; i++) {
            currentSum = 0;
            for (int j = i; j < i + 3; j++) {
                currentSum += depths.get(j);
            }
            if (currentSum > prev) {
                result++;
            }
            prev = currentSum;
        }
        return result;
    }

}

File: src/aoc2021/Day1/DepthIncreaseFinder.py
from aoc2021.utils import file_utils


def get_increased_depth_count(input):
    result = 0
    for i in range(len(input) - 1):
        if input[i + 1] > input[i]:
            result += 1
    return result


def get_increased_window_depth_count(input):
    result = 0
    prev_sum = 0

    for i in range(len(input) - 3):
        sum = 0
        j = i
        while j < i + 3:
            sum += int(input[j])
            j += 1

        if sum > prev_sum:
            result += 1
        prev_sum = sum

    return result


if __name__ == '__main__':
    input_path = "C:\\MyDrive\\Study\\My_Github_Repo\\Advent-of-Code\\src\\aoc2021\\Day1\\input1";
    input = file_utils.read_file(input_path)
    response = get_increased_depth_count(input)
    print(response)
    response = get_increased_window_depth_count(input)
    print(response)

    input_path = "C:\\MyDrive\\Study\\My_Github_Repo\\Advent-of-Code\\src\\aoc2021\\Day1\\input2";
    input = file_utils.read_file(input_path)
    response = get_increased_depth_count(input)
    print(response)
    response = get_increased_window_depth_count(input)
    print(response)
